### ISSUE TITLE
Implement `Least Recently Used` eviction in `RequestCache` .

### ISSUE THREAD

**Issue by happy-san** [typesense/typesense-dart#110]
### Is your feature request related to a problem? Please describe.
[`_cachedResponses`](https://github.com/typesense/typesense-dart/blob/96db58e96498b88a9af0bdad12067f0708a35719/lib/src/services/request_cache.dart#L9) doesn't have a limit on the number of `_Cache`'s it can store and the expired `_Cache` is [conditionally deleted](https://github.com/typesense/typesense-dart/blob/96db58e96498b88a9af0bdad12067f0708a35719/lib/src/services/request_cache.dart#L20), so a memory leak occurs.

### Describe the solution you'd like
Implement an LRU algorithm.

### Additional context
 [related issue](https://github.com/typesense/typesense-js/issues/66)

**Comment by happy-san** [comment]
Assigned to @harisarang

**Comment by happy-san** [comment]
@harisarang Maybe read up about [TLRU](https://en.wikipedia.org/wiki/Cache_replacement_policies#Time_aware_least_recently_used_(TLRU))- Time aware LRU scheme. I think TLRU scheme would be more relevant because LRU doesn't take `Time to utilize (TLU)` of the cached data into consideration.

**Comment by CoryADavis** [comment]
Is this library considered safe for production use without this fix?

**Comment by happy-san** [comment]
@CoryADavis I wouldn't recommend setting `cachedSearchResultsTTL` in production code yet.

**Comment by CoryADavis** [comment]
Ahh, understood, this issue is only related to a feature that is off by default.

Thank you!

**Comment by happy-san** [comment]
@o-artebiakin No, there's no way to clear the cache manually at the moment. Are you planning to use this feature in your app?
